thirdparty/jira.go

```diff
@@ -237,11 +277,24 @@ func (jiraHandler *JiraHandler) JQLSearchAll(query string) ([]JiraTicket, error)
 
 }
 
-func NewJiraHandler(server string, user string, password string) JiraHandler {
+func NewJiraHandler(opts send.JiraOptions) JiraHandler {
+	httpClient := http.DefaultClient
```

Comment:
can you use to pooled client?

Reply:
I did try to do that initially, but it made things weird with how the grip interfaces would be. I think we can either make grip allow more freedom for the callers to specify their own client (and maybe move the other methods here into grip) or leave as is - do you have a preference?

Reply:
grip/send.JiraOptions makes it possible to set a client. Just pull one from the pool.

Reply:
The problem is that because this struct is kept around in memory, we would never put the client back in the pool, so it made it use a temporary client to not have a resource leak

Reply:
can we add a close method on JiraHandler to do the put?

Reply:
That won't help - we still keep it around as long as the UI server is up

Reply:
I'd still rather pass one in rather than use the default one?

Reply:
This will mean we always have 1 fewer client in the pool - are you sure that's a good idea?

Reply:
yeah, that's fine. the clients are cheap enough, and if they're actually being used that's fine. Truthfully, I just want to make sure that the client gets configured in the same way as all the other clients, and DefaultClient use could conflict with libraries/etc. which might be a problem.

Reply:
fixed